deployer/src/proxy.rs

```diff
@@ -36,41 +40,27 @@ pub async fn handle(
     });
     span.set_parent(parent_context);
 
-    let host: FQDN = match req.headers().get(HOST) {
-        Some(host) => host
-            .to_str()
-            .unwrap_or_default()
-            .parse::<FQDN>()
-            .unwrap_or_default()
-            .to_owned(),
-        None => {
-            trace!("proxy request has no host header");
+    // First try to get the Host header. This header is not allowed in H2, so if we
+    // can't find it we try to get the host subcomponent from the URI authority.
+    let Some(host) = req
+        .headers()
+        .typed_get::<Host>()
+        .map(|host| fqdn!(host.hostname()))
+        .or_else(|| req.uri().host().map(|host| fqdn!(host))) else {
+            trace!("proxy request has no host header or URI authority host subcomponent");
             return Ok(Response::builder()
                 .status(StatusCode::BAD_REQUEST)
-                .body(Body::empty())
+                .body(Body::from("request has no host header or URI authority host subcomponent"))
                 .unwrap());
-        }
-    };
+        };
 
-    if host != fqdn {
```

Comment:
So the reason I removed this check is that for http2 requests that are proxied via gateway, the URI will be the gateway address and there won't be a host header. Therefore the host will not match the fqdn of the deployer.

What I'm wondering is if there is a problem with removing this? Will anyone outside of the controller be able to reach this proxy? :thinking:  Additionally, we do still check that the `XShuttleProject` header is here.